### ISSUE TITLE
Regenerate the npm package

### ISSUE THREAD

**Issue by Triztian** [google/protobuf#2028]
The current npm package does not have the `export.Map` statements that were added in this commit: https://github.com/google/protobuf/commit/ea081fe220ab15b9f449d8bfd77162085c24b73b.

Also in the process of verifying the root cause I found out that invoking `gulp dist`fails with the following message:

```
[21:55:23] Using gulpfile ~/Projects/protobuf/js/gulpfile.js
[21:55:23] Starting 'genproto_wellknowntypes'...

sh: ../src/protoc: No such file or directory

[21:55:23] 'genproto_wellknowntypes' errored after 25 ms
[21:55:23] Error: Command failed: sh -c ../src/protoc --js_out=import_style=commonjs,binary:. -I ../src ../src/google/protobuf/any.proto ../src/google/protobuf/api.proto ../src/google/protobuf/compiler/plugin.proto ../src/google/protobuf/descriptor.proto ../src/google/protobuf/duration.proto ../src/google/protobuf/empty.proto ../src/google/protobuf/field_mask.proto ../src/google/protobuf/source_context.proto ../src/google/protobuf/struct.proto ../src/google/protobuf/timestamp.proto ../src/google/protobuf/type.proto ../src/google/protobuf/wrappers.proto
sh: ../src/protoc: No such file or directory

    at ChildProcess.exithandler (child_process.js:202:12)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:850:16)
    at Socket.<anonymous> (internal/child_process.js:323:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at Pipe._handle.close [as _onclose] (net.js:492:12)
```

This can be fixed by executing the command like so: `env PROTOC=$(which protoc) gulp dist`
(Assuming that the `protoc` binary has been installed) the `protoc` variable inside `gulpfile.js` is set [here](https://github.com/google/protobuf/blob/master/js/gulpfile.js#L9) (Line 9).

The README should be updated to reflect that invocation option (though it's mentioned in the Setup section but not quite specific)

**Comment by xfxyjwf** [comment]
@haberman @pherl Does this need to be fixed in 3.0.2 patch release?

**Comment by haberman** [comment]
The `gulp dist` command is intended to be run from our source tree when we build the npm package. There is no reason to run it when you are installing *from* an npm package.
